evm/contracts/core/02-client/IBCClient.sol

```diff
@@ -1,135 +1,83 @@
-pragma solidity ^0.8.23;
+pragma solidity ^0.8.27;
 
-import "@openzeppelin/utils/Strings.sol";
 import "./ILightClient.sol";
 import "../25-handler/IBCMsgs.sol";
 import "../24-host/IBCStore.sol";
 import "../24-host/IBCCommitment.sol";
 import "../02-client/IIBCClient.sol";
-import "../../proto/ibc/core/client/v1/client.sol";
 
 library IBCClientLib {
-    event ClientRegistered(string clientType, address clientAddress);
-    event ClientCreated(string clientId);
-    event ClientUpdated(string clientId, IbcCoreClientV1Height.Data height);
+    event ClientRegistered(bytes32 clientType, address clientAddress);
+    event ClientCreated(uint32 clientId);
+    event ClientUpdated(uint32 clientId, uint64 height);
 
     error ErrClientTypeAlreadyExists();
-    error ErrClientMustNotBeSelf();
     error ErrClientTypeNotFound();
-    error ErrFailedToCreateClient();
-    error ErrFailedToUpdateClient();
 }
 
 /**
  * @dev IBCClient is a contract that implements [ICS-2](https://github.com/cosmos/ibc/tree/main/spec/core/ics-002-client-semantics).
  */
-contract IBCClient is IBCStore, IIBCClient {
+abstract contract IBCClient is IBCStore, IIBCClient {
     /**
      * @dev registerClient registers a new client type into the client registry
      */
     function registerClient(
-        string calldata clientType,
+        bytes32 clientType,
```

Comment:
is this basically just a string? padded with 0x00

Reply:
It would rather be the keccak256 of the string directly